src/server.ts

```diff
@@ -267,6 +256,24 @@ class TotalCount
     advisoryCount: number = 0;
 };
 
+const constructPayload = (packages, aggregator) => {
+    return new Promise((resolve) => {
+        let requestPayload = [];
+        for (let pck of packages) {
+            if (pck.name.value && pck.version.value && regexVersion.test(pck.version.value.trim())
+                && !(requestPayload.some((item) => item.package === pck.name.value && item.version === pck.version.value))) {

```

Comment:
What is the usage of `!(requestPayload.some((item)..` check?

Reply:
I used this to aggregate for each dependencies because for batch call the versions like "^1.2.3" are not passing the RegExp test.
But here can we aggregate requests based on request data?